packages/test/mocha-test-setup/src/mochaHooks.ts

```diff
@@ -4,28 +4,51 @@
  */
 
 import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { Context } from "mocha";
 
 const _global: any = global;
-const nullLogger: ITelemetryBufferedLogger = { send: () => {}, flush: async () => {} };
+class TestLogger implements ITelemetryBufferedLogger {
+    send(event: ITelemetryBaseEvent) {
+        event.testName = this.testName;
+        this.parentLogger.send(event);
+    }
+    async flush() {
+        return this.parentLogger.flush();
+    }
+    pass(title: string | undefined) {
+        this.testName = title;
+    }
+    constructor(private readonly parentLogger: ITelemetryBufferedLogger,
+        private testName: string | undefined) {}
```

Comment:
make test name optional
```suggestion
        private testName?: string) {}
```